src/template/template.ts

```diff
@@ -51,11 +51,19 @@ export default abstract class Template {
         return await namespaceDetector.getNamespace();
     }
 
+    private _getEolSetting(): string {
+        const eolSetting = vscode.workspace.getConfiguration().get('files.eol', EOL);
+
+        if (eolSetting !== '\n' && eolSetting !== '\r\n')
```

Comment:
What I don't get is: do we expect more than 3 values? Why don't make it explicit?

```javascript
if (eolSetting !== 'auto') {
  return EOL;
}
```
Ok, the above check will lead to introduce one more case for \n and \r\n. Which I feel cumbersome. 
Probably it might be better to use a `switch` where all the case are explicit. It won't introduce a different behavior but it will make the code more readable. 

```javascript
switch (eolSetting)
  case '\n':
  case '\r\n':
     return eolSetting;
  case 'auto':
  default:
     return EOL;
}

```
I assume that "auto" is the corresponding value for "auto" entry, in case it won't be and it doesn't exist a real mapping. Please ignore my comment.

Reply:
The value can be set to anything in the json itself, and vscode could add more values.
A switch case catches all, and is much more readable & explicit. I'll change that.

Reply:
I also thought that some more complex tests should be introduced to cover such feature. Not really aware if it's possible, but I'll glad to write them.

Reply:
Changed & tested a bit